myldr/boot.c

```diff
@@ -204,6 +264,66 @@ typedef BOOL (WINAPI *pALLOW)(DWORD);
     /* extract embedded_files[0] (i.e. the custom Perl interpreter) 
      * into stmpdir (but under the same basename as argv[0]) */
     my_prog = par_findprog(argv[0], par_getenv("PATH"));
+
+#ifdef __MACH__
+    /* Detect if FAT binary */
+    FILE *obj_file = fopen(my_prog, "rb");
+    uint32_t magic = read_magic(obj_file, 0);
+    fclose(obj_file);
+    if (magic == FAT_CIGAM || magic == FAT_MAGIC) {
+
+      /* Create seperate dir for extracted thin binary*/
```

Comment:
You can't call `par_mktmpdir` a second time (it has global side effects). But at this point you can just use `stmpdir`.

Reply:
Ah, yes, I wasn't sure about that. I was originally using `stmpdir` but this seemed not to work as the thin binary, on the second pass in, was being overwritten by the extraction of the custom perl, as mentioned in the code (it extracts to argv[0] name)? The thin binary needs to be argv[0] in the second pass so that the headers are checked.

Reply:
This is addressed in the latest commit